### PR TITLE
Add simple documentation how to use c/image with podman's rootless mode

### LINKED ISSUE #1400: Document how to access rootless containers-storage:
I am currently trying to read images from the local podman container storage via the following code:
```go
package main

import (
	"fmt"

	"github.com/containers/image/v5/transports/alltransports"
)

func main() {
	imgRef, err := alltransports.ParseImageName("containers-storage:registry.opensuse.org/opensuse/busybox:latest")
	if err != nil {
		panic(err)
	}
	fmt.Println(imgRef)
}
```
which fails with:
```
panic: chown /home/dan/.local/share/containers/storage/overlay/l: operation not permitted
```

Running this through strace reveals that the library tries to `chown` the directory to `root:root`, which fails when run as a normal user
```
fchownat(AT_FDCWD, "/home/dan/.local/share/containers/storage/overlay/l", 0, 0, 0) = -1 EPERM (Operation not permitted)
```

Now, given that skopeo can inspect that reference (`skopeo inspect containers-storage:registry.opensuse.org/opensuse/busybox`) leads me to the assumption that I am doing something wrong. Unfortunately, I don't really see what.

### PR DESCRIPTION
**Pull request by dcermak**
this fixes #1400

### CODE REVIEW

**Review by vrothberg — COMMENTED** [review]
LGTM
@mtrmac @giuseppe @lsm5 PTAL

**Comment by vrothberg** [comment]
Thanks, @dcermak.  I order to pass CI you need to sign your commit (e.g., via `git commit --amend -s`) and repush.

**Comment by dcermak** [comment]
Valentin Rothberg ***@***.***> writes:

> Thanks, @dcermak.  In order to pass CI you need to sign your commit (e.g., via `git commit --amend -s`) and repush.

Should be fixed now. Apologies, I missed that part completely.

**Review by giuseppe — APPROVED** [review]
LGTM

File: doc.go
Comment (mtrmac) [inline doc.go:35]: Something like “If your application needs to access a containers/storage store in rootless mode”, please; we don’t want the more common users of c/image/docker to do any of this.

File: doc.go
Comment (mtrmac) [inline doc.go:63]: The `"github.com/containers/storage/pkg/unshare"` import should be added to the example.

**Review by mtrmac — COMMENTED** [review]
Thanks! Sure, this place works well enough.

**Comment by TomSweeneyRedHat** [comment]
@dcermak thanks for the PR.  It needs an update and LGTM once @mtrmac 's comments are addressed.

File: doc.go
Comment (dcermak) [reply doc.go:63]: Right, I missed that. Thanks for spotting that!

File: doc.go
Comment (dcermak) [reply doc.go]: Makes sense, I've adjusted the wording accordingly.

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Review by vrothberg — APPROVED** [review]
LGTM

**Pull request merged**
